src/Catrobat/Controller/Web/ProgramController.php

```diff
@@ -506,6 +506,19 @@ public function translateProjectAction(Request $request, string $id): Response
     ]);
   }
 
+  /**
+   * @Route("/translate/custom/project/{id}", name="project_custom_translation", methods={"PUT"})
+   */
+  public function projectCustomTranslationAction(Request $request, string $id): Response
```

Comment:
Since you define the "PUT" method via a Symfony Routing annotation it is not possible that this action is called by a different method. You can get rid of the switch case.

Reply:
So the plan is to have the same endpoint handle PUT, GET, DELETE request, thus the switch on methods